test/stress/document_stress_test.go

```diff
@@ -0,0 +1,159 @@
+// +build stress
+
+/*
+ * Copyright 2020 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package stress
+
+import (
+	"fmt"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/yorkie-team/yorkie/api/converter"
+	"github.com/yorkie-team/yorkie/pkg/document"
+	"github.com/yorkie-team/yorkie/pkg/document/json"
+	"github.com/yorkie-team/yorkie/pkg/document/proxy"
+	"github.com/yorkie-team/yorkie/pkg/document/time"
+	"github.com/yorkie-team/yorkie/test/helper"
+)
+
+func TestDocumentStress(t *testing.T) {
+	t.Run("garbage collection test 2", func(t *testing.T) {
+		size := 10_000
+
+		// 01. initial
+		doc := document.New("c1", "d1")
+		fmt.Println("-----Initial")
+		bytes, err := converter.ObjectToBytes(doc.RootObject())
+		assert.NoError(t, err)
+		helper.PrintSnapshotBytesSize(bytes)
+		helper.PrintMemStats()
+
+		// 02. 10,000 integers
+		err = doc.Update(func(root *proxy.ObjectProxy) error {
+			root.SetNewArray("1")
+			for i := 0; i < size; i++ {
+				root.GetArray("1").AddInteger(i)
+			}
+
+			return nil
+		}, "sets big array")
+		assert.NoError(t, err)
+
+		// 03. deletes integers
+		err = doc.Update(func(root *proxy.ObjectProxy) error {
+			root.Delete("1")
+			return nil
+		}, "deletes the array")
+		assert.NoError(t, err)
+		fmt.Println("-----Integers")
+		bytes, err = converter.ObjectToBytes(doc.RootObject())
+		assert.NoError(t, err)
+		helper.PrintSnapshotBytesSize(bytes)
+		helper.PrintMemStats()
+
+		// 04. after garbage collection
+		assert.Equal(t, size+1, doc.GarbageCollect(time.MaxTicket))
```

Comment:
I guess we shouldn't call GarbageCollect() directly from this test package. GarbageCollect() should be private. It seems that the test code can be modified by indirectly calling it using the synchronization of the client. What do you think?

Reply:
It would be nice to change the `GarbageCollect` function to private. In this PR, I'd like to take care of separating stress tests and deal it with other issues later.